lib/core/src/system/gpu/data/texture.rs

```diff
@@ -427,31 +509,32 @@ pub struct BoundTexture<I,T> {
 /// Texture bound to GL context.
 #[derive(Debug)]
 pub struct BoundTextureData<I,T> {
-    texture    : Texture<I,T>,
     gl_texture : WebGlTexture,
     context    : Context,
+    format     : PhantomData<I>,
+    elem_type  : PhantomData<T>,
 }
 
 impl<I,T> BoundTextureData<I,T> {
     /// Constructor.
-    pub fn new(texture:Texture<I,T>,context:&Context) -> Self {
-        let gl_texture = context.create_texture().unwrap();
-        let context    = context.clone();
-        Self {texture,gl_texture,context}
+    pub fn new(context:&Context) -> Self {
+        Self {
+            gl_texture : context.create_texture().unwrap(),
+            context    : context.clone(),
+            format     : PhantomData,
+            elem_type  : PhantomData,
+        }
     }
-}
 
-impl<I:InternalFormat,T:TextureItemType> BoundTextureData<I,T> {
     /// Initializes default texture value. It is useful when the texture data needs to be downloaded
     /// asynchronously. This method creates a mock 1px x 1px texture and uses it as a mock texture
     /// until the download is complete.
     pub fn init_mock(&self) {
-        let texture         = &self.texture;
         let target          = Context::TEXTURE_2D;
         let level           = 0;
-        let internal_format = texture.gl_internal_format();
-        let format          = texture.gl_format().into();
-        let elem_type       = texture.gl_elem_type();
+        let internal_format = Context::RGBA as i32;
+        let format          = Context::RGBA;
+        let elem_type       = Context::UNSIGNED_BYTE;
```

Comment:
I don't like it. These parameters should never be provided manually. They should always be infered from our magic table like they were before.

Reply:
Ok, but that is actually a fix for this function. Below you created a RGBA data without checking what the actual format is.

And for me the final solution should be rather, that you don't specify format and elem_type while create UnitTexture.